customlist_data_builder.py

```diff
@@ -0,0 +1,67 @@
+import json
+import requests
+
+
+def build(employee, guide_id, zenefits_app_key):
+    """
+    Builds the CustomList data structure that will be
+    send to Builder.  The includes the following fields:
+        1) Import ID - the employee's ID used in Zenefits
+        2) Name - The employee's preferred name and last name
+        3) Title - The employee's title
+        4) Work Email
+        5) Work Phone
+        6) Location - which location they work out of
+        7) Department
+    """
+
+    # Cache the Locations and Departments so they don't have to
+    # be fetched over and over from Zenefits
+    location_cache = {}
```

Comment:
Reading through this again I realized this won't cache properly as it's written, I think this should be the approach:
in this file, wrap this build function in a class
```python
class CustomlistDataBuilder(object):
     def __init__(self, guide_id, zenefits_app_key):
        self.location_cache = {}
        self.department_cache = {}
        self.guide_id = guide_id
        self.zenefits_app_key = zenefits_app_key
    def build(self, employee):
        ...
    def get_zenefits_info(self, resource_url):
        ...
```

This way, in `data_loader.py` you can do:
```python
# outside the while loop:
customlist_builder = CustomlistDataBuilder(guide_id, zenefits_app_key)
...

# inside the while loop:
customlist_data = customlist_data_builder.build(employee)
...
```
This would ensure that for a given instantiation, you won't need to re-fetch any departments  or locations you've seen already. Wrapping it up in a class like that also means you can pass fewer arguments around for the `.build()` method calls. Would you agree with this approach?

Reply:
Sure, that makes sense.  I'll make those changes, thanks for the suggestion!